src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationManifestKindValueProvider.cs

```diff
@@ -0,0 +1,128 @@
+            {
+                return CustomManifestValue;
+            }
+
+            string noManifestPropertyValue = await defaultProperties.GetEvaluatedPropertyValueAsync(NoManifestMSBuildProperty);
+            if (noManifestPropertyValue?.Equals("true", StringComparison.InvariantCultureIgnoreCase) == true)
+            {
+                return NoManifestValue;
+            }
+
+            // It doesn't matter if it is set to false or the value is not present. We default to "DefaultManifest" scenario.
+            return DefaultManifestValue;
+        }
+
+        /// <summary>
+        /// Sets the application manifest kind property
+        /// </summary>
+        public override async Task<string?> OnSetPropertyValueAsync(
+            string propertyName,
+            string? unevaluatedPropertyValue,
+            IProjectProperties defaultProperties,
+            IReadOnlyDictionary<string, string>? dimensionalConditions = null)
+        {
+            if (string.Equals(unevaluatedPropertyValue, DefaultManifestValue, StringComparison.InvariantCultureIgnoreCase))
+            {
+                await SaveCurrentApplicationManifestValueAsync(defaultProperties);
+
+                await defaultProperties.DeletePropertyAsync(ApplicationManifestMSBuildProperty);
+                await defaultProperties.DeletePropertyAsync(NoManifestMSBuildProperty);
+            }
+            else if (string.Equals(unevaluatedPropertyValue, NoManifestValue, StringComparison.InvariantCultureIgnoreCase))
+            {
+                await SaveCurrentApplicationManifestValueAsync(defaultProperties);
+
+                await defaultProperties.DeletePropertyAsync(ApplicationManifestMSBuildProperty);
+                await defaultProperties.SetPropertyValueAsync(NoManifestMSBuildProperty, "true");
+            }
+            else if (string.Equals(unevaluatedPropertyValue, CustomManifestValue, StringComparison.InvariantCultureIgnoreCase))
+            {
+                await RestoreApplicationManifestValueAsync(defaultProperties);
+
+                await defaultProperties.DeletePropertyAsync(NoManifestMSBuildProperty);
+            }
+
+            // We don't want to store a value for this so return null.
+            return null;
+        }
+
+        /// <summary>
+        /// Save the current value of the "ApplicationManifest" MSBuild property, if any.
+        /// </summary>
+        private async Task SaveCurrentApplicationManifestValueAsync(IProjectProperties defaultProperties)
+        {
+            string? currentApplicationManifestPropertyValue = await defaultProperties.GetUnevaluatedPropertyValueAsync(ApplicationManifestMSBuildProperty);
+            if (!string.IsNullOrEmpty(currentApplicationManifestPropertyValue))
+            {
+                _temporaryPropertyStorage.AddOrUpdatePropertyValue(ApplicationManifestMSBuildProperty, currentApplicationManifestPropertyValue!);
+            }
```

Comment:
Using `Strings` here avoids the need for `!`.

```suggestion
            if (!Strings.IsNullOrEmpty(currentApplicationManifestPropertyValue))
            {
                _temporaryPropertyStorage.AddOrUpdatePropertyValue(ApplicationManifestMSBuildProperty, currentApplicationManifestPropertyValue);
            }
```